chain/starknet/src/adapter.rs

```diff
@@ -1,27 +1,219 @@
-use graph::blockchain::{EmptyNodeCapabilities, TriggerFilter as TriggerFilterTrait};
+use std::collections::{hash_map::Entry, HashMap};
+
+use graph::{
+    blockchain::{EmptyNodeCapabilities, TriggerFilter as TriggerFilterTrait},
+    components::store::BlockNumber,
+    firehose::{
+        BlockHeaderOnly as FirehoseFilterBlockHeaderOnly, BlockRange as FirehoseFilterBlockRange,
+        ContractEventFilter as FirehoseFilterContractEventFilter,
+        TopicWithRanges as FirehoseFilterTopicWithRanges,
+        TransactionEventFilter as FirehoseFilterTransactionEventFilter,
+    },
+};
+use prost::Message;
+use prost_types::Any;
+
+const BLOCK_HEADER_ONLY_TYPE_URL: &str =
+    "type.googleapis.com/zklend.starknet.transform.v1.BlockHeaderOnly";
+const TRANSACTION_EVENT_FILTER_TYPE_URL: &str =
+    "type.googleapis.com/zklend.starknet.transform.v1.TransactionEventFilter";
 
 use crate::{
     data_source::{DataSource, DataSourceTemplate},
+    felt::Felt,
     Chain,
 };
 
+type TopicWithRanges = HashMap<Felt, Vec<BlockRange>>;
```

Comment:
This type is quite opaque in how it's supposed to work, what does Felt represent here? Elsewhere it seems to be an address, here seems to be a "topic". 

Can you please make sure the Felt type comment covers what is represents? 

How many topics do we expect to represent in these? HashMap is probably overkill if it's only handling a handful of values.

Reply:
Oh sorry for the confusion. `Felt` is the universal primitive type in Starknet. Everything is a `Felt`: from addresses to contract call params to event logs, literally everything. I will make sure to cover this fact in the `Felt` type comment itself.

As for "topic", probably bad naming but what it means is the first element in any event `keys`, similar to Ethereum event signatures (which are the first element of any log "topics"). I should probably name this `signature` instead?

Reply:
maybe something like EventPrefix or EventFirstKey something along those lines, signature could make sense too if that's how it's treated, I think adding rust comments to the types will make that clearer regardless of the name :) I'm terrible at naming things eheh